pkg/gofr/gofr.go

```diff
@@ -309,87 +311,102 @@ func (a *App) initTracer() {
 
 	traceExporter := a.Config.Get("TRACE_EXPORTER")
 	tracerURL := a.Config.Get("TRACER_URL")
+	authHeader := a.Config.Get("TRACER_AUTH_KEY")
 
-	// deprecated : tracer_host and tracer_port is deprecated and will be removed in upcoming versions
+	// deprecated : tracer_host and tracer_port are deprecated and will be removed in upcoming versions.
 	tracerHost := a.Config.Get("TRACER_HOST")
 	tracerPort := a.Config.GetOrDefault("TRACER_PORT", "9411")
 
-	if tracerURL != "" && traceExporter == "" {
-		a.Logger().Error("missing TRACE_EXPORTER config, should be provided with TRACER_URL to enable tracing")
+	if traceExporter == "" {
+		a.Logger().Errorf("missing TRACE_EXPORTER config, should be provided with TRACER_URL to enable tracing")
+
 		return
 	}
 
-	//nolint:revive // early-return is not possible here, as below is the intentional logging flow
-	if tracerURL == "" && traceExporter != "" && !strings.EqualFold(traceExporter, "gofr") {
-		if tracerHost != "" && tracerPort != "" {
-			a.Logger().Warn("TRACER_HOST and TRACER_PORT are deprecated, use TRACER_URL instead")
-		} else {
-			a.Logger().Error("missing TRACER_URL config, should be provided with TRACE_EXPORTER to enable tracing")
-			return
-		}
+	if tracerHost == "" && tracerURL == "" && traceExporter != traceExporterGoFr {
+		a.Logger().Errorf("missing TRACER_URL config, should be provided with TRACE_EXPORTER to enable tracing")
+		return
 	}
 
-	if (traceExporter != "" && tracerHost != "") || tracerURL != "" || traceExporter == gofrTraceExporter {
-		exporter, err := a.getExporter(traceExporter, tracerHost, tracerPort, tracerURL)
+	if tracerURL == "" && tracerHost != "" {
+		a.Logger().Warn("TRACER_HOST and TRACER_PORT are deprecated, use TRACER_URL instead")
 
-		if err != nil {
-			a.container.Error(err)
-		}
+		tracerURL = fmt.Sprintf("%s:%s", tracerHost, tracerPort)
```

Comment:
URL should not be formed here.. as this URL is only specific for otlp.. we are updating the URL again for zipkin and gofr, lines: 384 and 400 respectively.
I believe this needs to be fixed

Reply:
host and port will be removed in later versions. For GofrTraceExporter, in my  latter commit  changes, gofr will only take effect on trace_url set or default , and the changes here to Zipkin will have no impact and even support  zipkin for url env set no only host and port .

Reply:
@Sianao - can we connect on discord? I have shared some idea, just wanted to check with you.